maildir.go

```diff
@@ -123,6 +123,24 @@ func (d Dir) UnseenCount() (int, error) {
 	return c, nil
 }
 
+func parseKey(filename string) string {
+	split := strings.FieldsFunc(filename, func(r rune) bool {
+		return r == separator
+	})
+
+	if len(split) == 0 {
+		return ""
+	}
+
+	return split[0]
+}
+
+// Key returns the key for the given file path.
```

Comment:
Maybe add "If the filepath doesn't refer to a message in the directory, an empty string is returned"?

Reply:
Or do you think it'd be more appropriate to return an error?

Reply:
Yeah on second thought I think returning an error is more appropriate instead of just failing silently. Added generic errors both here and if the maildir and path differ, but I can replace them with named errors if you want.